packages/bridge-ui/src/components/form/BridgeForm.svelte

```diff
@@ -2,30 +2,87 @@
   import { _ } from "svelte-i18n";
   import { token } from "../../store/token";
   import { fromChain, toChain } from "../../store/chain";
-  import { activeBridge, chainIdToBridgeAddress } from "../../store/bridge";
+  import {
+    activeBridge,
+    chainIdToBridgeAddress,
+    bridgeType,
+  } from "../../store/bridge";
   import { signer } from "../../store/signer";
   import { BigNumber, ethers, Signer } from "ethers";
   import { toast } from "@zerodevx/svelte-toast";
+  import type { Token } from "../../domain/token";
+  import type { BridgeType } from "../../domain/bridge";
+  import type { Chain } from "../../domain/chain";
 
   let amount: string;
+  let requiresAllowance: boolean = true;
   let btnDisabled: boolean = true;
 
   $: isBtnDisabled($signer, amount)
     .then((d) => (btnDisabled = d))
     .catch((e) => console.log(e));
 
+  $: checkAllowance(amount, $token, $bridgeType, $fromChain, $signer)
+    .then((a) => (requiresAllowance = a))
+    .catch((e) => console.log(e));
+
+  async function checkAllowance(
+    amt: string,
+    token: Token,
+    bridgeType: BridgeType,
+    fromChain: Chain,
+    signer: Signer
+  ) {
+    if (!signer || !amt || !token || !fromChain) return true;
+
+    return await $activeBridge.RequiresAllowance({
+      amountInWei: amt
+        ? ethers.utils.parseUnits(amt, token.decimals)
+        : BigNumber.from(0),
+      signer: signer,
+      contractAddress: token.address,
+      spenderAddress: $chainIdToBridgeAddress.get(fromChain.id),
+    });
+  }
+
   async function isBtnDisabled(signer: Signer, amount: string) {
     if (!signer) return true;
     if (!amount) return true;
+    if (requiresAllowance) return true;
     const balance = await signer.getBalance("latest");
     if (balance.lt(ethers.utils.parseUnits(amount, $token.decimals)))
       return true;
 
     return false;
   }
 
+  async function approve() {
+    try {
+      if (!requiresAllowance)
+        throw Error("does not require additional allowance");
+
+      const tx = await $activeBridge.Approve({
+        amountInWei: ethers.utils.parseUnits(amount, $token.decimals),
+        signer: $signer,
+        contractAddress: $token.address,
+        spenderAddress: $chainIdToBridgeAddress.get($fromChain.id),
+      });
+      console.log("approved, waiting for confirmations ", tx);
+      $signer.provider.waitForTransaction(tx.hash, 3);
```

Comment:
```suggestion
      await $signer.provider.waitForTransaction(tx.hash, 3);
```
`await` is needed since `waitForTransaction` returns a `Promise`?